src/libraries/System.Collections.Immutable/src/System/Runtime.InteropServices/ImmutableCollectionsMarshal.cs

```diff
@@ -54,5 +55,19 @@ public static ImmutableArray<T> AsImmutableArray<T>(T[]? array)
         {
             return array.array;
         }
+
+        /// <summary>
+        /// Gets a <see cref="Memory{T}"/> for the <typeparamref name="T"/> array underlying an input <see cref="ImmutableArray{T}.Builder"/>.
+        /// </summary>
+        /// <typeparam name="T">The type of elements in the input <see cref="ImmutableArray{T}.Builder"/> value.</typeparam>
+        /// <param name="builder">The builder.</param>
+        /// <returns>
+        /// A <see cref="Memory{T}"/> for the filled portion of <typeparamref name="T"/> array underlying
+        /// the input <see cref="ImmutableArray{T}.Builder"/>.
+        /// </returns>
+        public static Memory<T> AsMemory<T>(ImmutableArray<T>.Builder? builder)
```

Comment:
Why is the input nullable?

Reply:
Same as CollectionsMarshal.AsSpan and ImmutableCollectionsMarshal.AsImmutableArray.

Reply:
That's the first time I notice. What's the motivation behind allowing null?

Reply:
Not sure with the existing ones. But either way we need a check, so it's a question of whether we want that failed check to throw or return the equivalent of empty. Seemed prudent to tension consistent with the others.